File: app/src/main/java/com/zuk0/gaijinsmash/riderz/ui/activity/splash/SplashActivity.java
package com.zuk0.gaijinsmash.riderz.ui.activity.splash;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

import com.zuk0.gaijinsmash.riderz.ui.activity.main.MainActivity;

import javax.inject.Inject;

import dagger.android.AndroidInjection;


public class SplashActivity extends AppCompatActivity {

    private SplashViewModel mViewModel;

    @Inject
    SplashViewModelFactory splashViewModelFactory;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        initDagger();
        super.onCreate(savedInstanceState);
        initViewModel();
        initStations();
        launchMainActivity();
    }

    private void initDagger() {
        AndroidInjection.inject(this);
    }

    private void initViewModel() {
        mViewModel = ViewModelProviders.of(this, splashViewModelFactory).get(SplashViewModel.class);
    }

    private void initStations() {
        mViewModel.initStationsData();
    }

    private void launchMainActivity() {
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/zuk0/gaijinsmash/riderz/ui/activity/splash/SplashActivity.kt
package com.zuk0.gaijinsmash.riderz.ui.activity.splash

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider

import com.zuk0.gaijinsmash.riderz.ui.activity.main.MainActivity

import javax.inject.Inject

import dagger.android.AndroidInjection
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasAndroidInjector


class SplashActivity : AppCompatActivity(), HasAndroidInjector {

    @Inject lateinit var fragmentInjector : DispatchingAndroidInjector<Any>

    //@Inject lateinit var splashViewModelFactory: ViewModelProvider.AndroidViewModelFactory
    private lateinit var vm: SplashViewModel

    override fun androidInjector(): AndroidInjector<Any> {
        return fragmentInjector
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        initDagger()
        super.onCreate(savedInstanceState)
        initViewModel()
        vm.initDayNightTheme()
        initStations()
        launchMainActivity()
    }

    private fun initDagger() {
        AndroidInjection.inject(this)
    }

    private fun initViewModel() {
        vm = ViewModelProvider(this).get(SplashViewModel::class.java)
    }

    //todo: need to move this to application
    private fun initStations() {
        vm.initStationsData()
    }

    private fun launchMainActivity() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
        finish()
    }
}
